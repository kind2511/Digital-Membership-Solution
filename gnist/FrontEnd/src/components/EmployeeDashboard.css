.emp-dashboard-container {
  display: flex;
  flex-direction: column;
  height: 100vh;
  overflow: hidden;
}

.emp-navbar {
  display: flex;
  justify-content: center; 
  align-items: center;
  background-color: RGB(19, 182, 253); 
  color: #FFFFFF;
  padding: 20px 0;
  box-shadow: 0 4px 6px rgba(0,0,0,0.1);
}

.emp-navbar-menu {
  display: flex;
  justify-content: space-around;
  list-style: none;
  padding: 0;
  margin: 0;
  width: 100%;
  max-width: 1200px;
}

.emp-nav-item {
  cursor: pointer;
  text-align: center;
  padding: 10px 20px;
  transition: color 0.3s;
  position: relative;
}

.emp-nav-item::after {
  content: '';
  position: absolute;
  bottom: -10px;
  left: 50%;
  transform: translateX(-50%);
  width: 0;
  height: 2px;
  background-color: #FFFFFF; 
  transition: width 0.3s;
}

.emp-nav-item:hover, .emp-nav-item.emp-active {
  color: #D9E3F0;
}

.emp-nav-item:hover::after, .emp-nav-item.emp-active::after {
  width: 100%;
}

.emp-main-content {
  flex-grow: 1;
  padding: 20px;
  background-color: #FAFAFA; 
  overflow: auto;
}

.emp-logout-item {
  color: #FAFAFA; 
}

@media (max-width: 768px) {
  .emp-navbar-menu {
    flex-direction: column;
  }
  
  .emp-nav-item {
    width: 100%;
    padding: 10px;
  }
  
  .emp-logout-item {
    padding-top: 15px;
  }
}
