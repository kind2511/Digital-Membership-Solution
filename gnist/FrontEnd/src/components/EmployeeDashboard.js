import React, { useState } from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import Tilstede from './Tilstede';
import Aktiviteter from './Aktiviteter';
import Undersøkelser from './Undersøkelser';
import Rød from './Rød';
import Medleminfo from './Medleminfo'; 
import './EmployeeDashboard.css';

function EmployeeDashboard() {
  const { logout } = useAuth0();
  const [activeNavItem, setActiveNavItem] = useState('Tilstede');
  const [showLogoutModal, setShowLogoutModal] = useState(false); 

  const handleLogout = () => {
    // Show logout confirmation modal
    setShowLogoutModal(true);
  };

  const handleLogoutConfirmation = () => {
    // Perform logout
    logout({ returnTo: window.location.origin });
  };

  const renderContent = () => {
    switch (activeNavItem) {
      case 'Tilstede':
        return <Tilstede />;
      case 'Rød':
        return <Rød />;
      case 'Aktiviteter':
        return <Aktiviteter />;
      case 'Undersøkelser':
        return <Undersøkelser />;
      case 'Medleminfo': 
        return <Medleminfo />;
      default:
        return <div className="emp-content">Please select an item from the navbar.</div>;
    }
  };

  return (
    <div className="emp-dashboard-container">
      <div className="emp-navbar">
        <div className="emp-navbar-menu">
          <div 
            className={`emp-nav-item ${activeNavItem === 'Tilstede' ? 'emp-active' : ''}`} 
            onClick={() => setActiveNavItem('Tilstede')}
          >
            Tilstede
          </div>
          <div 
            className={`emp-nav-item ${activeNavItem === 'Rød' ? 'emp-active' : ''}`} 
            onClick={() => setActiveNavItem('Rød')}
          >
            Rød
          </div>
          <div 
            className={`emp-nav-item ${activeNavItem === 'Aktiviteter' ? 'emp-active' : ''}`} 
            onClick={() => setActiveNavItem('Aktiviteter')}
          >
            Aktiviteter
          </div>
          <div 
            className={`emp-nav-item ${activeNavItem === 'Undersøkelser' ? 'emp-active' : ''}`} 
            onClick={() => setActiveNavItem('Undersøkelser')}
          >
            Undersøkelser
          </div>
          <div 
            className={`emp-nav-item ${activeNavItem === 'Medleminfo' ? 'emp-active' : ''}`} 
            onClick={() => setActiveNavItem('Medleminfo')}
          >
            Medleminfo
          </div>
          <div 
            className="emp-nav-item emp-logout-item" 
            onClick={handleLogout}
          >
            Logg Ut
          </div>
        </div>
      </div>
      <div className="emp-main-content">
        {renderContent()}
      </div>
      {showLogoutModal && (
        <div className="logout-modal">
          <div className="logout-modal-content">
            <p>Er du sikker på at du vil logge ut?</p>
            <div className="logout-modal-buttons">
              <button className="logout-confirm-button" onClick={handleLogoutConfirmation}>Ja</button>
              <button className="logout-confirm-button" onClick={() => setShowLogoutModal(false)}>Nei</button>
            </div>
          </div>
        </div>
      )}
    </div>
  );
}

export default EmployeeDashboard;
