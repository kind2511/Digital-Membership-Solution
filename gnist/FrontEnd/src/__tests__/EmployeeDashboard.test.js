import React from 'react';
import renderer from 'react-test-renderer';
import EmployeeDashboard from '../components/EmployeeDashboard';
import { Auth0Context } from '@auth0/auth0-react';

// Mock the child components
jest.mock('../components/Tilstede', () => () => <div>Tilstede Component</div>);
jest.mock('../components/Aktiviteter', () => () => <div>Aktiviteter Component</div>);
jest.mock('../components/Undersøkelser', () => () => <div>Undersøkelser Component</div>);
jest.mock('../components/Rød', () => () => <div>Rød Component</div>);
jest.mock('../components/Medleminfo', () => () => <div>Medleminfo Component</div>);
jest.mock('../components/FinnBruker', () => () => <div>FinnBruker Component</div>);

// Mock Auth0 Context
const mockAuth0Context = {
  user: { sub: 'auth0|123456789' },
  logout: jest.fn()
};

test('EmployeeDashboard component snapshot', () => {
  const component = renderer.create(
    <Auth0Context.Provider value={mockAuth0Context}>
      <EmployeeDashboard />
    </Auth0Context.Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
