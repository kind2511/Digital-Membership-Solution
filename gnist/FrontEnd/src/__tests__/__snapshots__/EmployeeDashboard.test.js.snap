// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeeDashboard component snapshot 1`] = `
<div
  className="emp-dashboard-container"
>
  <div
    className="emp-navbar"
  >
    <div
      className="emp-navbar-menu"
    >
      <div
        className="emp-nav-item emp-active"
        onClick={[Function]}
      >
        Tilstede
      </div>
      <div
        className="emp-nav-item "
        onClick={[Function]}
      >
        Rød
      </div>
      <div
        className="emp-nav-item "
        onClick={[Function]}
      >
        Aktiviteter
      </div>
      <div
        className="emp-nav-item "
        onClick={[Function]}
      >
        Undersøkelser
      </div>
      <div
        className="emp-nav-item "
        onClick={[Function]}
      >
        Medleminfo
      </div>
      <div
        className="emp-nav-item "
        onClick={[Function]}
      >
        Finn Bruker
      </div>
      <div
        className="emp-nav-item emp-logout-item"
        onClick={[Function]}
      >
        Logg Ut
      </div>
    </div>
  </div>
  <div
    className="emp-main-content"
  >
    <div>
      Tilstede Component
    </div>
  </div>
</div>
`;
